integration-libs/opf/base/root/components/opf-payment-verification/opf-payment-verification.service.ts

```diff
@@ -0,0 +1,269 @@
+    ) {
+      return of(true);
+    } else if (response.result === OpfPaymentVerificationResult.CANCELLED) {
+      return throwError({
+        ...this.defaultError,
+        message: 'opf.payment.errors.cancelPayment',
+      });
+    } else {
+      return throwError(this.defaultError);
+    }
+  }
+
+  displayError(error: HttpErrorModel | undefined): void {
+    this.globalMessageService.add(
+      {
+        key:
+          error?.message && error?.status === -1
+            ? error.message
+            : 'opf.payment.errors.proceedPayment',
+      },
+      GlobalMessageType.MSG_TYPE_ERROR
+    );
+  }
+
+  checkIfProcessingCartIdExist(): void {
+    this.opfService
+      .getOpfMetadataState()
+      .pipe(
+        take(1),
+        filter(
+          (opfPaymentMetadata: OpfPaymentMetadata) =>
+            opfPaymentMetadata.isPaymentInProgress === false
+        )
+      )
+      .subscribe(() => {
+        this.goToPage(OpfPage.CART_PAGE);
+
+        this.globalMessageService.add(
+          {
+            key: 'httpHandlers.cartNotFound',
+          },
+          GlobalMessageType.MSG_TYPE_ERROR
+        );
+      });
+  }
+
+  runHostedPagePattern(paymentSessionId: string, paramsMap: KeyValuePair[]) {
+    return this.verifyPayment(paymentSessionId, paramsMap).pipe(
+      concatMap(() => {
+        return this.placeOrder();
+      }),
+      map((order) => !!order),
+      tap((success: boolean) => {
+        if (success) {
+          this.goToPage(OpfPage.CONFIRMATION_PAGE);
+        }
+      })
+    );
+  }
+
+  runHostedFieldsPattern(
+    domain: GlobalFunctionsDomain,
+    paymentSessionId: string,
+    vcr: ViewContainerRef,
+    paramsMap: Array<KeyValuePair>
+  ): Observable<boolean> {
+    this.globalFunctionsService.registerGlobalFunctions({
+      domain,
+      paymentSessionId,
+      vcr,
+      paramsMap,
+    });
+
+    return this.opfPaymentService.afterRedirectScripts(paymentSessionId).pipe(
+      concatMap((response) => {
+        if (!response?.afterRedirectScript) {
+          return throwError(this.defaultError);
+        }
+        return from(
+          this.renderAfterRedirectScripts(response.afterRedirectScript)
+        );
+      })
+    );
+  }
+
+  protected renderAfterRedirectScripts(
+    script: OpfDynamicScript
+  ): Promise<boolean> {
+    const html = script?.html;
+
+    return new Promise((resolve: (value: boolean) => void) => {
+      this.opfResourceLoaderService
+        .loadProviderResources(script.jsUrls, script.cssUrls)
+        .then(() => {
+          if (html) {
+            this.opfResourceLoaderService.executeScriptFromHtml(html);
+            resolve(true);
+          } else {
+            resolve(false);
+          }
+        })
+        .catch(() => {
+          resolve(false);
+        });
+    });
+  }
+
+  removeResourcesAndGlobalFunctions(): void {
+    this.globalFunctionsService.removeGlobalFunctions(
+      GlobalFunctionsDomain.REDIRECT
```

Comment:
[MEDIUM]

Here we remove the global function domain using the *hardcoded* value `GlobalFunctionsDomain.REDIRECT`, but in the same service, we don't register it, using the same *hardcoded* value. Therefore, I guess we delegate this responsibility to another place. My intuition is that we should use this hardcoded value in only one class, so only one class is responsible for deciding how to use it.

For example, I can see that we pass it as a parameter from the component to the service:
```runHostedFieldsPattern(domain: GlobalFunctionsDomain)```
Perhaps we could drop this parameter and just hardcode it in this service as well